### PR TITLE
[frontend|backend]Show prerequisites info in atomic testing

### PR DESCRIPTION
**Pull request by johanah29**
#1856

### CODE REVIEW

File: openbas-api/src/main/java/io/openbas/executors/Injector.java [hunk 1]
```diff
@@ -54,7 +55,7 @@ public void setFileService(FileService fileService) {
   public abstract ExecutionProcess process(Execution execution, ExecutableInject injection)
       throws Exception;
 
-  public InjectStatusCommandLine getCommandsLines(String externalId) {
+  public PayloadOutput getPayloadOutput(String externalId) {
     return null;
   }
```
Comment (savacano28) [inline openbas-api/src/main/java/io/openbas/executors/Injector.java:60]: @damgouj Why are we returning null here (we pass a param but it's never used)? Couldn't we at least use Optional?

**Review by heditar — COMMENTED** [review]
Few comments about the code structure, I didn't test the code yet

File: openbas-api/src/main/java/io/openbas/executors/Injector.java (hunk 1 above)
Comment (johanah29) [reply openbas-api/src/main/java/io/openbas/executors/Injector.java:60]: This method is used to translate the third parties systems  approach of payloads  into openBAS payloads modeling. For now it is only implemented for Caldera.

File: openbas-api/src/main/java/io/openbas/utils/AtomicTestingUtils.java [hunk 2]
```diff
@@ -567,4 +578,104 @@ public record ExpectationResultsByType(
 
   public record ResultDistribution(
       @NotNull String id, @NotNull String label, @NotNull Integer value) {}
+
+  public PayloadOutputDto getPayloadOutputFromInject(Optional<Inject> inject) {
+    PayloadOutputDto.PayloadOutputDtoBuilder payloadOutputDtoBuilder = PayloadOutputDto.builder();
```
Comment (savacano28) [inline openbas-api/src/main/java/io/openbas/utils/AtomicTestingUtils.java:583]: I prefer to move this method to PayloadMapper.

**Review by heditar — APPROVED** [review]

File: openbas-api/src/main/java/io/openbas/rest/atomic_testing/form/PayloadOutputDto.java [hunk 3]
```diff
@@ -0,0 +1,90 @@
...
+import lombok.Builder;
+import lombok.Getter;
+import lombok.Setter;
+import lombok.Singular;
+
+@Setter
+@Getter
+@Builder
+public class PayloadOutputDto {
+
```
Comment (savacano28) [inline openbas-api/src/main/java/io/openbas/rest/atomic_testing/form/PayloadOutputDto.java:20]: I'd prefer just use: output, input, simple

File: openbas-api/src/main/java/io/openbas/utils/AtomicTestingUtils.java [hunk 4]
```diff
@@ -567,4 +578,104 @@ public record ExpectationResultsByType(
...
+              .executableArch(injectorContract.getArch());
+          if (COMMAND_TYPE.equals(injectorContract.getPayload().getType())) {
+            Command payloadCommand = (Command) Hibernate.unproxy(payload);
+            List<PayloadCommandBlock> payloadCommandBlocks = new ArrayList<>();
+            PayloadCommandBlock payloadCommandBlock =
+                new PayloadCommandBlock(
+                    payloadCommand.getExecutor(),
+                    payloadCommand.getContent(),
+                    List.of(payloadCommand.getCleanupCommand()));
+            payloadCommandBlocks.add(payloadCommandBlock);
```
Comment (savacano28) [inline openbas-api/src/main/java/io/openbas/utils/AtomicTestingUtils.java:611]: We have some list.of(null)

File: openbas-api/src/main/java/io/openbas/utils/InjectUtils.java [hunk 5]
```diff
@@ -1,70 +1,166 @@
...
-                    && !payloadCommand.getCleanupCommand().isBlank()
-                ? List.of(payload.getCleanupCommand())
-                : null,
-            payload.getExternalId());
+        PayloadCommandBlock payloadCommandBlock =
+            new PayloadCommandBlock(
+                payloadCommand.getExecutor(),
+                payloadCommand.getContent(),
+                List.of(payloadCommand.getCleanupCommand()));
+        return new PayloadOutput(
```
Comment (savacano28) [inline openbas-api/src/main/java/io/openbas/utils/InjectUtils.java:49]: Validate getCleanupCommand is not null

File: openbas-api/src/main/java/io/openbas/utils/InjectUtils.java [hunk 6]
```diff
@@ -1,70 +1,166 @@
...
+            null);
       } else {
-        try {
-          // Inject comes from Caldera ability and tomorrow from other(s) Executor(s)
-          io.openbas.execution.Injector executor =
-              context.getBean(
-                  injectorContract.getInjector().getType(), io.openbas.execution.Injector.class);
-          return executor.getCommandsLines(injectorContract.getId());
-        } catch (NoSuchBeanDefinitionException e) {
-          log.info(
```
Comment (savacano28) [inline openbas-api/src/main/java/io/openbas/utils/InjectUtils.java:59]: Why delete the try catch?

File: openbas-front/src/utils/Localization.js [hunk 7]
```diff
@@ -2508,7 +2516,19 @@ const i18n = {
...
+      'Commands': '命令',
+      'Executable files': '可执行文件',
+      'File drop file': '文件放置文件',
+      'Dns resolution hostname': 'Dns 解析主机名',
+      'Network traffic ip destination': '网络流量 IP 目标',
+      'Network traffic port destination': '网络流量端口目标',
+      'Network traffic ip source': '网络流量 IP 源',
+      'Network traffic port source': '网络流量端口源',
+      'Network traffic protocol': '网络流量协议',
+
```
Comment (savacano28) [inline openbas-front/src/utils/Localization.js:2530]: delete 2530l

File: openbas-model/src/main/java/io/openbas/database/model/AttackPatternSimple.java [hunk 8]
```diff
@@ -0,0 +1,29 @@
...
+
+import com.fasterxml.jackson.annotation.JsonProperty;
+import jakarta.validation.constraints.NotBlank;
+import lombok.Getter;
+import lombok.Setter;
+
+@Setter
+@Getter
+public class AttackPatternSimple {
+
```
Comment (savacano28) [inline openbas-model/src/main/java/io/openbas/database/model/AttackPatternSimple.java:11]: This class is an entity?

**Review by savacano28 — CHANGES_REQUESTED** [review]
Some errors frontend side

**Pull request closed without merging**